### PR TITLE
Refactor structure of package updates functions

### PR DESCRIPTION
**Pull request by amcaplan**
### WHY are these changes introduced?

The TODO says it all. We should have a straightforward "upgrade without thinking" function which can be called by an "upgrade only certain things" function, not the other way around.

I meant to do this some time ago, especially given it broke `shopify upgrade` right at the Summer Edition launch, but I never got back to it. UNTIL NOW.

### WHAT is this pull request doing?

Refactors the update code to have functions' dependencies point in the right direction.

### How to test your changes?

Run `shopify upgrade` or generate an extension on an existing project.

### Measuring impact

How do we know this change was effective? Please choose one:

- [x] n/a - this doesn't need measurement, e.g. a linting rule or a bug-fix
- [ ] Existing analytics will cater for this addition
- [ ] PR includes analytics changes to measure impact

### Checklist

- [x] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [x] I've considered possible [documentation](https://shopify.dev) changes

### CODE REVIEW

File: packages/cli-kit/src/node/node-package-manager.ts [hunk 1]
```diff
@@ -334,7 +335,7 @@ ${token.json(options)}
       args = argumentsToAddDependenciesWithPNPM(depedenciesWithVersion, options.type)
       break
   }
-  options.stdout?.write(`Executing...${args.join(' ')}`)
+  options.stdout?.write(`Executing... ${[options.packageManager, ...args].join(' ')}`)
```
Comment (amcaplan) [inline packages/cli-kit/src/node/node-package-manager.ts:338]: This line is not strictly related to the PR topic, but the output seems much more helpful.

File: packages/cli-main/src/cli/commands/upgrade.ts [hunk 2]
```diff
@@ -86,10 +85,10 @@ export default class Upgrade extends Command {
     const packagesToUpdate = packages.filter((pkg: string): boolean => {
       const pkgRequirement: string | undefined = deps[pkg]
       return Boolean(pkgRequirement)
-    })
+    }).map((pkg) => { return {name: pkg, version: 'latest'} })
```
Comment (amcaplan) [inline packages/cli-main/src/cli/commands/upgrade.ts:88]: This needs to be updated because the method accepts an array of `DependencyVersion`s, not strings.

File: packages/cli-kit/src/node/node-package-manager.ts [hunk 3]
```diff
@@ -298,7 +298,6 @@ export interface DependencyVersion {
 export async function addNPMDependenciesIfNeeded(
   dependencies: DependencyVersion[],
   options: AddNPMDependenciesIfNeededOptions,
-  force = false,
```
Comment (amcaplan) [inline packages/cli-kit/src/node/node-package-manager.ts:301]: This `force` param is the gross hack that I wanted to get around. `IfNeeded` means this is a more specialized method, and we shouldn't have a `force` param which contradicts the `IfNeeded`.

**Review by isaacroldan — APPROVED** [review]
